### ISSUE TITLE
[Bug] NumPy and `bfloat16`

### ISSUE THREAD

**Issue by jaketae** [unifyai/ivy#9492]
While examining why test cases were failing for 

```
pytest ivy_tests/test_ivy/test_frontends/test_torch/test_creation_ops.py -k 'test_torch_full'
```

I realized that the `to_numpy()` function has a special branch for handling `bfloat16`:

https://github.com/unifyai/ivy/blob/e4f6cb774f18b8789c99a1ac4df22f342e858764/ivy/functional/backends/torch/general.py#L86-L92

The line I'm struggling to parse is 

```python
return x.detach().cpu().numpy().astype("bfloat16")
```

Currently, NumPy has no support for `bfloat16`. This can easily be checked in the interpreter.

```
>>> import numpy as np
>>> x = np.random.randn(1)
>>> x.astype("float16")
array([-0.343], dtype=float16)
>>> x.astype("bfloat16")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: data type 'bfloat16' not understood
```

There is an [ongoing discussion](https://github.com/numpy/numpy/issues/19808) in the NumPy repo regarding `bfloat16` support. Currently, they are conservative with adopting `bfloat16` and are waiting for an external proof-of-concept implementation from the community. 

In short, how are we handling `bfloat16` in the `to_numpy()` conversion? Some related PRs I've found so far:

* #6362
* #6364
* #6366
* #6476

Thank you!

**Comment by jaketae** [comment]
I found some PRs that appeared relevant to this topic.

* #6362
* #6364
* #6366
* #6476

@aditya-pola @VedPatwardhan, do you have any insight you could share? Thank you!
